### PR TITLE
End tile fix

### PR DESCRIPTION
**Pull request by gknisely**
fixes the mass/boston core dump.

### CODE REVIEW

File: src/mjolnir/transitbuilder.cc [hunk 1]
```diff
@@ -379,18 +379,18 @@ void AddOSMConnection(const Transit_Stop& stop, const GraphTile* tile,
         if (directededge->use() == Use::kFerry || directededge->use() == Use::kRailFerry) {
           float start_distance = node->latlng().Distance(stop_ll);
 
-          const DirectedEdge* opp_de = tile->directededge(node->edge_index() + directededge->opp_index());
-          GraphId end_node = opp_de->endnode();
-          float end_distance = 0.0f;
-
           const GraphTile* endnode_tile = tile;
-          if ( directededge->endnode().Tile_Base() != end_node.Tile_Base()) {
+          if ( directededge->endnode().Tile_Base() != tile->id()) {
```
Comment (dnesbitt61) [inline src/mjolnir/transitbuilder.cc]: Tile_Base I think is tileid and level, so not sure about comparing to just id

Comment (gknisely) [inline src/mjolnir/transitbuilder.cc]: @dnesbitt61 Are you sure?  graphenhancer.cc line 405 and 1200 has this same logic.

Comment (dnesbitt61) [inline src/mjolnir/transitbuilder.cc]: On second look you are right. The tile->id() method returns the GraphId stored in the header, which should just include the tile Id and level.

Comment (gknisely) [inline src/mjolnir/transitbuilder.cc]: @dnesbitt61 ok.  Also added end_tile check in validator.

File: src/mjolnir/graphvalidator.cc [hunk 2]
```diff
@@ -79,6 +79,10 @@ uint32_t GetOpposingEdgeIndex(const GraphId& startnode, DirectedEdge& edge,
 		uint64_t wayid, const GraphTile* tile, const GraphTile* end_tile,
 		uint32_t& dupcount, std::string& endnodeiso) {
 
+  if (!end_tile) {
```
Comment (dnesbitt61) [inline src/mjolnir/graphvalidator.cc]: Do you ever see this?

**Pull request merged**